distributed/deploy/tests/test_spec_cluster.py

```diff
@@ -83,6 +83,7 @@ def test_spec_sync(loop):
 
 
 @pytest.mark.filterwarnings("ignore:There is no current event loop:DeprecationWarning")
+@pytest.mark.filterwarnings("ignore:make_current is deprecated:DeprecationWarning")
 def test_loop_started_in_constructor(cleanup):
     # test that SpecCluster.__init__ starts a loop in another thread
     cluster = SpecCluster(worker_spec, scheduler=scheduler, loop=None)
```

Comment:
AIUI, this is explicitly deprecated functionality (building a `LoopRunner` without an active loop), but unpicking that seems like a larger job.

Reply:
building a LoopRunner without an active loop is not deprecated - LoopRunner will use `asyncio.run` to start a new loop in the future

Reply:
Perhaps I misunderstand the current deprecation warnings then:

https://github.com/dask/distributed/blob/main/distributed/utils.py#L435-L457

Right now, if we don't pass a `loop` object, and ask for an asynchronous `LoopRunner` and there is not asyncio loop running we get a deprecation warning. So presumably this branch should disappear (if `asynchronous=True` one must pass a loop object).

If no loop is passed and we're not asynchronous, then the idea is that we get a new `IOLoop` for a thread (eventually). In this case, I suppose we must build a new loop with `asyncio.new_event_loop()`? AIUI, `asyncio.run` will build a new loop, run a coroutine, and then tear the loop down, which doesn't _seem_ like what is desired here.

Contrarily, if a `loop` _is_ provided and the wrapped asyncio event loop is not running, we get a deprecation warning. So presumably this should become unsupported behaviour now.

Reply:
> In this case, I suppose we must build a new loop with asyncio.new_event_loop()? AIUI, asyncio.run will build a new loop, run a coroutine, and then tear the loop down, which doesn't seem like what is desired here.

yes using asyncio.run in this way is the plan, the change in `LoopRunner._start_unlocked.run_loop` will look like:

```python
        def run_loop(loop=self._loop):
            async def _():
                self._stop_event = asyncio.Event()
                self._loop = IOLoop.current()
                loop_cb()
                await self._stop_event.wait()

            try:
                asyncio.run(_())
            except Exception as e:
                start_exc[0] = e
            finally:
                done_evt.set()
```